src/main/scala/io/pravega/connectors/spark/PravegaMicroBatchStream.scala

```diff
@@ -142,6 +143,13 @@ class PravegaMicroBatchStream(
     } else {
       val upperLimit = limits.find(_.isInstanceOf[ReadMaxRows]).map(_.asInstanceOf[ReadMaxRows])
       PravegaSourceOffset(batchClientFactory.getNextStreamCut(startOffset, upperLimit.get.maxRows()))
+      try {
+        PravegaSourceOffset(batchClientFactory.getNextStreamCut(startOffset, upperLimit.get.maxRows()))
+      }
+      catch
+      {
+        case e: SegmentTruncatedException => PravegaSourceOffset(PravegaUtils.getStreamInfo(streamManager, scopeName, streamName).getTailStreamCut)
+      }
     }
```

Comment:
In case of SegmentTruncatedException falling back to original logic of read all by passing tailStreamCut.

Reply:
It's not correct, it should recalculate from the current head, go to the tail would result in data loss.

Reply:
Updated to pass headstreamcut.